tests/meta/transfer/test_external_data.py

```diff
@@ -0,0 +1,210 @@
+        raw_path="/path/%s", customer=customer_id, lims_sample_id=sample_id
+    )
+
+    # THEN the source path is
+    assert destination_path == Path("/path/cust000/ADM1/")
+
+
+def test_download_sample(
+    external_data_api: ExternalDataAPI,
+    mocker,
+    ticket_nr: int,
+    customer_id: str,
+    cust_sample_id: str,
+    sample_id: str,
+):
+    """Test for downloading external data via slurm"""
+
+    # GIVEN paths needed to run rsync
+    mocker.patch.object(ExternalDataAPI, "create_log_dir")
+    ExternalDataAPI.create_log_dir.return_value = Path("/path/to/log")
+
+    mocker.patch.object(ExternalDataAPI, "create_source_path")
+    ExternalDataAPI.create_source_path.return_value = Path("/path/to/source")
+
+    mocker.patch.object(ExternalDataAPI, "create_destination_path")
+    ExternalDataAPI.create_destination_path.return_value = Path("/path/to/destination")
+
+    # WHEN the destination path is created
+    sbatch_number = external_data_api.transfer_sample(
+        cust_sample_id=cust_sample_id,
+        ticket_id=ticket_nr,
+        cust=customer_id,
+        lims_sample_id=sample_id,
+        dry_run=True,
+    )
+
+    # THEN check that an integer was returned as sbatch number
+    assert isinstance(sbatch_number, int)
+
+
+def test_get_all_fastq(
+    cg_context: CGConfig, external_data_directory, external_data_api: ExternalDataAPI
+):
+    """Test the finding of fastq.gz files in customer directories."""
+    for folder in external_data_directory.iterdir():
+        # WHEN the list of file-paths is created
+        files = external_data_api.get_all_fastq(
+            sample_folder=external_data_directory.joinpath(folder)
+        )
+        # THEN only fast.gz files are returned
+        assert [tmp.suffixes == [".fastq", ".gz"] for tmp in files]
+
+
+def test_check_bundle_fastq_files(
+    external_data_api: ExternalDataAPI, sample_id: str, hk_version_obj
+):
+
+    # GIVEN a housekeeper bundle with a file
+    # mocker.patch.object(MockHousekeeperAPI, "get_files")
+    # MockHousekeeperAPI.get_files.return_value = [Path("/boring/old/boomer/file.fastq.gz")]
+    external_data_api.housekeeper_api.add_file(
+        path=Path("/boring/old/boomer/file.fastq.gz"), tags=["fastq"], version_obj=hk_version_obj
+    )
+
+    # WHEN when attempting to add two files, one existing and one new
+    files_to_add: List[Path] = external_data_api.check_bundle_fastq_files(
+        fastq_paths=[Path("/wow/new/cool/file.fastq.gz"), Path("/boring/old/boomer/file.fastq.gz")],
+        lims_sample_id=sample_id,
+        last_version=hk_version_obj,
+    )
+
+    # Then only the new file should be returned
+    assert files_to_add == [Path("/wow/new/cool/file.fastq.gz")]
+
+
+def test_add_files_to_bundles(
+    external_data_api: ExternalDataAPI, sample_id: str, hk_version_obj, fastq_file: Path
+):
+    """Tests adding files to housekeeper and checking corresponding md5file."""
```

Comment:
```suggestion
    """Tests adding files to housekeeper and checking corresponding md5file"""
```
We should standardize the docs strings in cg. Right now it there is no best practise....